pkg/client/unversioned/request.go

```diff
@@ -722,10 +757,8 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			// TCP connection
 			defer resp.Body.Close()
 
-			retries++
-			if seconds, wait := checkWait(resp); wait && retries < maxRetries {
-				glog.V(4).Infof("Got a Retry-After %s response for attempt %d to %v", seconds, retries, url)
-				time.Sleep(time.Duration(seconds) * time.Second)
+			if seconds, wait := checkWait(resp); wait {
+				glog.V(4).Infof("Got a Retry-After %s response for attempt %d to %v, but will backoff exponentially instead.", seconds, url)
```

Comment:
In a single master the retry interval is good but the  loadbalanaced scenarios which are more realistic, the retry has diminishing value.  For example if  overloaded and x% of requests are failing, backing off exponentially gaurantees that the a request swarm doesn't hit LBs or individual endpoint which are swamped .

Cc @timothysc